### PR TITLE
feat: Wrap/Unwrap Key API

### LINKED ISSUE #65: Implement Wrap/Unwrap Key API

### PR DESCRIPTION
**Pull request by aholovko**
Closes #65

Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: pkg/kms/service.go [hunk 1]
```diff
@@ -179,6 +184,63 @@ func (s *service) VerifyMAC(keystoreID, keyID string, mac, data []byte) error {
 	return nil
 }
 
+// WrapKey wraps cek for the recipient with public key 'recipientPubKey'.
+func (s *service) WrapKey(keystoreID, keyID string, cek, apu, apv []byte,
+	recipientPubKey *crypto.PublicKey) (*crypto.RecipientWrappedKey, error) {
+	if keyID != "" {
+		kh, err := s.getKeyHandle(keystoreID, keyID)
+		if err != nil {
+			return nil, err
```
Comment (DRK3) [inline pkg/kms/service.go:193]: Looks like there are some unwrapped errors in this file

File: pkg/restapi/kms/operation/operations.go [hunk 2]
```diff
@@ -438,6 +584,51 @@ func (o *Operation) writeResponse(rw io.Writer, v interface{}) {
 	}
 }
 
+func unmarshalPublicKey(k *publicKey) (*crypto.PublicKey, error) {
```
Comment (DRK3) [inline pkg/restapi/kms/operation/operations.go:587]: Unwrapped errors

**Review by DRK3 — APPROVED** [review]

File: pkg/kms/service.go (hunk 1 above)
Comment (aholovko) [reply pkg/kms/service.go:193]: I am trying to wrap only once when code passes package boundary - in this case getKeyHandle already does the wrapping.

File: pkg/restapi/kms/operation/operations.go (hunk 2 above)
Comment (aholovko) [reply pkg/restapi/kms/operation/operations.go:587]: thanks, this case is more subtle - will think on this in my next PR

File: pkg/kms/service.go [hunk 3]
```diff
@@ -179,6 +184,63 @@ func (s *service) VerifyMAC(keystoreID, keyID string, mac, data []byte) error {
...
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	if senderPubKey != nil {
+		senderKH, e := keyio.PublicKeyToKeysetHandle(senderPubKey)
+		if e != nil {
+			return nil, NewServiceError(unwrapKeyFailed, e)
+		}
```
Comment (Baha-sk) [inline pkg/kms/service.go:225]: this conversation is unnecessary

`crypto.WithSender()`
now accepts `*crypto.PublicKey` directly

Comment (aholovko) [reply pkg/kms/service.go:225]: Thanks @Baha-sk, I will update this in my next PR.

**Pull request merged**